src/cxy_ctl.erl

```diff
@@ -601,9 +601,26 @@ execute_wrapper(Mod, Fun, Args, Task_Type, Max_History, Start, Spawn_Or_Inline,
     end.
 
 -spec fail_wrapper(spawn | inline, any(), any()) -> no_return().
-fail_wrapper(spawn,  Call_Data, Stacktrace) -> erlang:error(spawn_failure,  [Call_Data, Stacktrace]);
-fail_wrapper(inline, Call_Data, Stacktrace) -> exit       ({inline_failure, [Call_Data, Stacktrace]}).
-
+fail_wrapper(spawn,  Call_Data, Stacktrace) -> erlang:error(spawn_failure,  [sanitize(Call_Data), Stacktrace]);
+fail_wrapper(inline, Call_Data, Stacktrace) -> exit       ({inline_failure, [sanitize(Call_Data), Stacktrace]}).
+
+% timeout on nested gen_server:call/2,3 - most callers are
+% gcache modules trying to replenish cache (create_key_value)
+% in the deepest callstack, redis connection goes away for a bit
+
+% search for 'spawn_failure' or 'inline_failure' in the logs, you will see more
+% pattern match for specific case - trade off between
+% valuable debugging information vs. private customer's data
+sanitize({exit, {timeout, {gen_server, call, [ServerRef, Request, Timeout]}},
+    {creating_new_value_with, gcache_auth,
+        {check_auth, {api_key, _ApiKey, _ApiSecret, CxyCtlOpts}}}}) ->
+
+    {exit, {timeout, {gen_server, call, [ServerRef, Request, Timeout]}},
+        {creating_new_value_with, gcache_auth,
+            {check_auth, {api_key, xxx, xxx, CxyCtlOpts}}}};
```

Comment:
can we please keep some part of the API key? E.g. several letters? That would help us debug which API has problem,

Reply:
@anderswei @nitzanharel several letters might not be useful. You can be mistaken with other api keys. Can we print api key but no secret? What is the rules?

Also, the core problem here is redis connection / wpool that holds the conn goes away / timeout. We better spend time to look deeper.

Dict_Props might do but requires changes at call sites, do it later? do we not want to sanitize it?

Reply:
The problem here is not always Redis. A timeout here can be because of endless reasons. Even if Redis is the problem, it might be because of things we can't control (e.g. network issues) - and we would still want to NOT show these things in the logs.

yes - it will require changes at the call side, but : 
1. it might end up being just one place because of how we reuse code
2. the sanitize code suggested have nothing to do with the library (epocxy) but are specific to usage
3. as we find more clauses - we will want to update this stuff and updating a library for that feels out of place

Reply:
As for the rule - I think that printing the api key without the password should be fine.